arangod/IResearch/IResearchLinkHelper.cpp

```diff
@@ -55,6 +55,39 @@
 
 namespace {
 
+bool isIgnoredHiddenEnterpriseCollection(std::string const& cName) {
+  /*
+   * Note: As IResearchView.cpp L204 says:
+   * -> "create links on a best-effort basis, link creation failure does not cause view creation failure"
+   *
+   * Workaround: If we detect a collection which should not be created in the SingleServer case,
+   * let the link validation itself return a success.
+   *
+   * Nevertheless, the user will be notified that there has been an edge case. This should be fine.
+   * Another attempt could be to rewrite the links itself, but more code changes will then be
+   * necessary.
+   */
+#ifdef USE_ENTERPRISE
+  if (arangodb::ServerState::instance()->isSingleServer()) {
+    if (cName[0] == '_') {
+      if (strncmp(cName.c_str(), arangodb::StaticStrings::FullLocalPrefix.c_str(), 7) == 0 ||
+          strncmp(cName.c_str(), arangodb::StaticStrings::FullFromPrefix.c_str(), 6) == 0 ||
+          strncmp(cName.c_str(), arangodb::StaticStrings::FullToPrefix.c_str(), 4) == 0) {
```

Comment:
It would be nice to have a short comment on the magical numbers 7, 6 and 4.

Reply:
ty done.